diplom.py

```diff
@@ -9,6 +9,25 @@ def parse(line):
     label = int (tokens[23])
     return (features, label)
 
+def mean_average_precision (predicted_probability_distribution, 
+    n_of_classes, 
```

Comment:
Do you need n_of_classes here? I think you can derive it from other arguments i.e. len(list_of_classes). Please also add an assertion i.e. "assert len(list_of_classes) == len(predicted_probability_distribution)".